packages/grafana-runtime/src/analytics/plugins/usePluginInteractionReporter.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  type DataSourcePluginContextType,
+  isDataSourcePluginContext,
+  type PluginContextType,
+  usePluginContext,
+} from '@grafana/data';
+
+import { reportInteraction } from '../utils';
+
+import { createDataSourcePluginEventProperties, createPluginEventProperties } from './eventProperties';
+
+export function usePluginInteractionReporter(): typeof reportInteraction {
+  const context = usePluginContext();
+
+  if (isDataSourcePluginContext(context)) {
+    return createDataSourceReporter(context);
+  }
+
+  return createPluginReporter(context);
+}
+
+function createDataSourceReporter(context: DataSourcePluginContextType): typeof reportInteraction {
+  const { meta, dataSource } = context;
+  const info = createDataSourcePluginEventProperties(meta, dataSource);
+
+  return (interactionName: string, properties?: Record<string, unknown>) => {
+    return reportInteraction(interactionName, { ...properties, ...info });
+  };
+}
+
+function createPluginReporter(context: PluginContextType): typeof reportInteraction {
+  const { meta } = context;
+  const info = createPluginEventProperties(meta);
+
+  return (interactionName: string, properties?: Record<string, unknown>) => {
+    return reportInteraction(interactionName, { ...properties, ...info });
```

Comment:
Is there a chance that we would like to override anything in the `info{}` object during reporting? If yes, would it make sense to shift around the destructuring assignment? Or would we like to be sure that these default values cannot be overriden?
```suggestion
    return reportInteraction(interactionName, { ...info, ...properties });
```

Reply:
This is actually done on purpose. The idea was that you should not be able to override the default values. Maybe we should throw some error if you try to do that?